pkg/controller/pravegacluster/upgrade.go

```diff
@@ -378,3 +425,18 @@ func (r *ReconcilePravegaCluster) getPodsWithVersion(sts *appsv1.StatefulSet, ve
 	}
 	return pods, nil
 }
+
+func checkUpgradeCondition(p *pravegav1alpha1.PravegaCluster, reason string, updatedReplicas int32) error {
+	_, lastCondition := p.Status.GetClusterCondition(pravegav1alpha1.ClusterConditionUpgrading)
+	if lastCondition.Reason == reason && lastCondition.Message != string(updatedReplicas) {
```

Comment:
Here we seem to be checking against time since cluster went into "upgrading" state.
That is not correct. We need to check against time since this sts went into upgrading state.

Reply:
This is not. Please check the last comment.